### PR TITLE
initial commit

### PR DESCRIPTION
**Pull request by AlexsisMoskovchuk**

### CODE REVIEW

File: checkstyle.xml [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+        <module name="EmptyCatchBlock">
+            <property name="exceptionVariableName" value="expected"/>
+        </module>
+        <module name="CommentsIndentation"/>
+        <module name="DeclarationOrder"/>
+        <module name="SingleSpaceSeparator">
+            <property name="validateComments" value="true"/>
+        </module>
+    </module>
+</module>
```
Comment (issugy) [inline checkstyle.xml]: add empty line

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,76 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>mate-academy</groupId>
+    <artifactId>diamond</artifactId>
```
Comment (issugy) [inline pom.xml]: add informative group and artifact ids

File: src/main/java/cinema/dao/impl/ShoppingCartDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+import cinema.model.ShoppingCart;
+import cinema.model.User;
+import cinema.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class ShoppingCartDaoImpl implements ShoppingCartDao {
+
```
Comment (issugy) [inline src/main/java/cinema/dao/impl/ShoppingCartDaoImpl.java]: redundant empty line

File: src/main/java/cinema/dao/impl/TicketDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+import cinema.lib.Dao;
+import cinema.model.Ticket;
+import cinema.util.HibernateUtil;
+import cinema.dao.TicketDao;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+
+@Dao
+public class TicketDaoImpl implements TicketDao {
+
```
Comment (issugy) [inline src/main/java/cinema/dao/impl/TicketDaoImpl.java]: redundant empty line

File: src/main/java/cinema/exception/AuthenticationException.java [hunk 5]
```diff
@@ -0,0 +1,8 @@
+package cinema.exception;
+
+public class AuthenticationException extends Exception {
+
```
Comment (issugy) [inline src/main/java/cinema/exception/AuthenticationException.java]: redundant empty line

File: src/main/java/cinema/dao/impl/UserDaoImpl.java [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(user);
+            transaction.commit();
+            return user;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't insert to DB user: "
```
Comment (issugy) [inline src/main/java/cinema/dao/impl/UserDaoImpl.java]: lets make messages of exceptions be consistent

File: src/main/java/cinema/model/MovieSession.java [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+@Entity
+@Table(name = "movie_sessions")
+public class MovieSession {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @ManyToOne(fetch = FetchType.LAZY)
+    private Movie movie;
+    @ManyToOne(fetch = FetchType.LAZY)
+    private CinemaHall cinemaHall;
```
Comment (issugy) [inline src/main/java/cinema/model/MovieSession.java:24]: add name for column

File: src/main/java/cinema/model/MovieSession.java [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+    private LocalDateTime showTime;
```
Comment (issugy) [inline src/main/java/cinema/model/MovieSession.java:26]: same

File: src/main/java/cinema/model/ShoppingCart.java [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+import javax.persistence.OneToOne;
+import javax.persistence.Table;
+
+@Entity
+@Table (name = "shopping_carts")
+public class ShoppingCart {
+    @Id
+    private Long id;
+    @OneToMany
+    private List<Ticket> tickets;
```
Comment (issugy) [inline src/main/java/cinema/model/ShoppingCart.java:22]: add names for table and columns

File: src/main/java/cinema/model/Ticket.java [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+import javax.persistence.ManyToOne;
+import javax.persistence.Table;
+
+@Entity
+@Table(name = "tickets")
+public class Ticket {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
```
Comment (issugy) [inline src/main/java/cinema/model/Ticket.java]: redundant empty line

File: src/main/java/cinema/model/Ticket.java [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+    @ManyToOne(fetch = FetchType.LAZY)
+    private MovieSession movieSession;
```
Comment (issugy) [inline src/main/java/cinema/model/Ticket.java:20]: add name for column

File: src/main/java/cinema/model/Ticket.java [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+
```
Comment (issugy) [inline src/main/java/cinema/model/Ticket.java]: redundant empty line

File: src/main/java/cinema/model/Ticket.java [hunk 13]
```diff
@@ -0,0 +1,55 @@
...
+        this.movieSession = movieSession;
+    }
+
+    @Override
+    public String toString() {
+        return "Ticket{" + "id=" + id
+              + ", movieSession=" + movieSession
+              + ", user=" + user + '}';
+    }
+
```
Comment (issugy) [inline src/main/java/cinema/model/Ticket.java]: redundant empty line

File: src/main/java/cinema/security/AuthenticationServiceImpl.java [hunk 14]
```diff
@@ -0,0 +1,42 @@
...
+
+    @Override
+    public User register(String email, String password) {
+        User user = new User();
+        user.setEmail(email);
+        user.setPassword(password);
+        userService.add(user);
+        shoppingCartService.registerNewShoppingCart(user);
+        return user;
+    }
```
Comment (issugy) [inline src/main/java/cinema/security/AuthenticationServiceImpl.java]: where do u throw registration exception ?

File: src/main/java/cinema/util/HashUtil.java [hunk 15]
```diff
@@ -0,0 +1,31 @@
+package cinema.util;
+
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+
+public class HashUtil {
```
Comment (issugy) [inline src/main/java/cinema/util/HashUtil.java:7]: add private default constructor

File: src/test/java/core/basesyntax/HelloWorldTest.java [hunk 16]
```diff
@@ -0,0 +1,8 @@
+package core.basesyntax;
+
+/**
+ * Feel free to remove this class and create your own.
+ */
+public class HelloWorldTest {
+
+}
```
Comment (issugy) [inline src/test/java/core/basesyntax/HelloWorldTest.java]: remove this file

**Review by issugy — CHANGES_REQUESTED** [review]

File: pom.xml [hunk 17]
```diff
@@ -4,7 +4,7 @@
          xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
     <modelVersion>4.0.0</modelVersion>
 
-    <groupId>mate-academy</groupId>
+    <groupId>cinema-app</groupId>
     <artifactId>diamond</artifactId>
```
Comment (issugy) [inline pom.xml]: what about artifact id ?

**Review by issugy — CHANGES_REQUESTED** [review]
fix all previous comments

File: pom.xml (hunk 17 above)
Comment (AlexsisMoskovchuk) [reply pom.xml]: fixed

**Review by issugy — APPROVED** [review]

**Pull request merged**